test-requirements.txt

```diff
@@ -177,6 +177,8 @@ types-docutils==0.21.0.20241128
     # via -r test-requirements.in
 types-pyopenssl==24.1.0.20240722
     # via -r test-requirements.in
+types-pyyaml==6.0.12.20250326
```

Comment:
I think it's probably better to actually add a dependency on whatever yaml import we want. It shouldn't actually matter because pre-commit will use it too, so we're already installing it.

Reply:
`pyyaml` is what `pre-commit` itself uses, which is also the reason I am using it. This is the type annotations package for mypy to be able to understand it without reporting errors in CI.

Reply:
Sorry for some reason I forgot that we don't have `pre-commit` as a dep in this file.